bin/run-command.sh

```diff
@@ -18,10 +18,71 @@ set -euo pipefail
 # TODO: Add ability to change task IAM Role. Part 3 of multipart update https://github.com/navapbc/template-infra/issues/354#issuecomment-1693973424
 # TODO: Change to keyword arguments. Part 3 of multipart update https://github.com/navapbc/template-infra/issues/354#issuecomment-1693973424
 
-APP_NAME="$1"
-ENVIRONMENT="$2"
-COMMAND="$3"
-ENVIRONMENT_VARIABLES=${4:-""}
+ACCOUNT_ID="$(./bin/current-account-id.sh)"
+
+while :; do
+  # echo "$1 $2"
+  if [ -z ${1+x} ]; then
+    break
+  fi
+  case $1 in
+    --environment-variables)
+      if [ "$2" ]; then
+        ENVIRONMENT_VARIABLES=$2
+        shift 2
+      else
+        echo "$1 flag present without Environment Variables"
+        exit 1
+      fi
+      ;;
+    --role-name)
+      if [ -z ${ROLE_ARN+x} ]; then
+        if [ "$2" ]; then
+          ROLE_ARN="arn:aws:iam::$ACCOUNT_ID:role/$2"
+          shift 2
+        else
+          echo "$1 flag present without role name"
+          exit 1
+        fi
+      else
+        printf "Conflicting attempts to define role\n%s\n%s" "$ROLE_ARN" "$1"
+        exit 1
+      fi
+      ;;
+    --role-arn)
+      if [ -z ${ROLE_ARN+x} ]; then
```

Comment:
this check is unnecessary. i don't see value in checking for the option being present multiple times. in fact even heavily used shell utilities have the behavior of treating the last option as the one that is respected. this allows people to compose things better.

Reply:
Sounds good. I'll drop those checks

Reply:
Resolved in e8bd95c